bot.py

```diff
@@ -40,6 +40,20 @@
 with open("knowledge/specific.json", "r", encoding='utf-8') as f:
     kattis_specific = json.load(f)
 
+deleted_messages = {} # {channel_id : message}
+
+@client.event
+async def on_message_delete(message):
+    deleted_messages[message.channel.id] = message
+
+@client.command(name='snipe')
+async def snipe(ctx):
+    message = deleted_messages.pop(ctx.channel.id, None)
```

Comment:
do we only want snipe to work on first invocation?

Reply:
we shouldn't maintain a list of deleted messages. some people may want to erase their old messages. the purpose for only sniping the last deleted message is for seeing what someone deleted right as it was deleted. it'd be even better to only maintain that message in this dict for just a few seconds.